build/documentation/copy-package-readme.js

```diff
@@ -26,39 +30,130 @@ process.argv.forEach(function (val, index) {
     }
 
     if (!validArg) {
-        console.log('Invalid argument used. To perform a dry-run use --dry-run');
+        console.log('\x1b[31m%s\x1b[0m', 'Invalid argument used. To perform a dry-run use --dry-run');
         process.exit(1);
     }
+
 });
 
 /**
- * Function to copy readme files to the docs/en/packages folder
+ * Run script
+ * Based off presence of --dry-run parameter
  */
-function copyReadmeFiles() {
+dryRun? runDryRun() : copyReadmeFiles();
+
+/**
+ * Function that describes what the ./copy-package-readme script will do and
+ * where, without making any actual changes
+ */
+function runDryRun() {
 
     const resolvedSrcReadmePaths = path.resolve(rootDir, srcReadmePaths);
 
-    glob(resolvedSrcReadmePaths, {realpath:true}, function(error, files) {
-        
-        files.forEach((filePath) => {
-            const destReadmePath = filePath.replace(/(\bpackages\b)(?!.*\1)/, destDir);
+    console.log("In docs/en/packages/, this script would...");
+
+    glob(resolvedSrcReadmePaths, {realpath:true}, function(error, srcFiles) {
+        srcFiles.forEach((srcReadmePath) => {    
```

Comment:
Sorry I should have been more clear in my previous comment. The main point I was trying to make is that the dry-run flag should still largely follow the same code-path as the actual function implementation, with the exception of not writing the files out. The opportunity still exists for these two functions to get out of sync and the "dry-run" to not accurately reflect what the functions do.

Instead, the dry-run should execute the exact same code as the non-dry-run, with the exception of writing the files.  If we're deriving the file-paths in the same code-path across both cases, we can be much more certain this function is doing what we expect.